Keeper tests

husio opened the pull request:
@alpe please check if this is what you had in mind. Two of those tests are failing right now.

alpe commented:
Yes, these are good findings. It does not mean that we have to implement the address validation in the keeper (as this is/ should be done downstream) but at least we can have proper documentation for people using the keeper directly. 👍

alpe reviewed (changes requested):
Thanks for the contribution 💐 
Please see my comment on the gzip test case that needs to be fixed.  Otherwise let's rebase and get this in.

husio commented on x/wasm/keeper/keeper_test.go:
```diff
@@ -238,18 +268,13 @@ func TestCreateWithGzippedPayload(t *testing.T) {
-	require.NoError(t, err)
-
-	contractID, err := keeper.Create(ctx, creator, wasmCode, nil)
+	contractID, err := keeper.Create(ctx, creator, hackatomWasm, nil)
```
That is a great catch! I can smell a find & replace mistake :see_no_evil:

husio commented on x/wasm/keeper/keeper_test.go:
```diff
@@ -23,35 +24,73 @@ import (
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 )
 
+// When migrated to go 1.16, embed package should be used instead.
```
Yes, this is a good idea. When using `embed` package is possible, static value declaration will be also clean and straightforward.

husio commented:
I have rebased, fixed merge conflict and squashed all changes into a single commit. Force push was a must.

husio commented:
I have added a check to the keeper that ensures `creator` address to not be `nil`. This probably happens higher up the stack as well, but keeper is able to do such check as well.

I have removed the test that is using an invalid address bytes. I assume in the keeper, there is no good way to check if address bytes are correct (validation and deserialization happens before).
